lib/gears/matcher.lua

```diff
@@ -170,6 +178,42 @@ function matcher:matches_rules(o, rules)
     return false
 end
 
+--- Assign a function to match an object property against a value.
+--
+-- The default matcher uses the `==` operator for all types. It also uses the
+-- `:match()` method for string and allows pattern matching.
```

Comment:
What happened to "functions are called"? The `awful.rules` configuration in the default config uses `focus = awful.client.focus.filter` where the later is a function that is called with a client and returns true/false.

Reply:
Still there.

I will add it do this documentation section (which is unrelated).